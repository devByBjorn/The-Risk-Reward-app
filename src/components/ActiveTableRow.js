import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
// import { calculatePositiveR, calculateNegativeR } from '../calculations/riskRewardCalculation'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteFirebaseTrade } from '../actions/tradeActions'

const ActiveTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    entry,
    stop,
    target,
    status,
    opened,
    negativeR,
    positiveR,
    id }) => (
    status === 'active' &&
    <tr>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{entry}</td>
      <td>{stop}</td>
      <td>{target}</td>
      <td>{opened}</td>
      <td>{negativeR}</td> {/* loss -R*/}
      <td>{positiveR}</td> {/* win +R*/}
      <td>
        <div>
          <Link to={`/edit-trade/${id}`}>
            <EditIcon />
          </Link>
          <button onClick={() => {
            dispatch(deleteFirebaseTrade({ id }))
          }}
          ><TrashIcon />
          </button>
        </div>
      </td>
    </tr>
  )

export default connect()(ActiveTableRow)