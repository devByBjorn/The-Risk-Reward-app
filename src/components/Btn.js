import React from 'react'

export const Btn = (props) => (
  <button
    type="button"
    {...props}
  >
    {props.text}</button>
)

export const SubmitBtn = (props) => (
  <button>{props.text}</button>
)

export default Btn