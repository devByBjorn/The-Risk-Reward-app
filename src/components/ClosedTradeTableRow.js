import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteTrade } from '../actions/tradeActions'




const ClosedTradeTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    opened,
    closed,
    status,
    outcome,
    rewardToRisk,
    id }) => {

  const [rowColor, setRowColor] = useState('')

  const colors = {
    win: '#7FBA00',
    loss: '#F0341B',
    scratch: '#FFB900'
  }
  const { win, loss, scratch } = colors

  useEffect(() => {
    if (outcome === 'win') {
      setRowColor(win)
    } else if (outcome === 'loss') {
      setRowColor(loss)
    } else {
      setRowColor(scratch)
    }
  })

  return (
    status === 'closed' &&

    <tr style={{ backgroundColor: rowColor }}>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{rewardToRisk}</td>
      <td className="desktop">{opened}</td>
      <td className="desktop">{closed}</td>
      <td className="desktop">{outcome}</td>
      <td className="handle-trade-row">
        <div className="handle-trade">
          <Link to={`/edit-trade/${id}`}>
            <EditIcon
              className="edit-icon icon"
            />
          </Link>
          <button
            className="btn-transparent"
            onClick={() => {
              alert('Do you want to delete this trade?')
              dispatch(deleteTrade({ id }))
            }}
          ><TrashIcon
              className="trashcan-icon icon"
            />
          </button>
        </div>
      </td>
    </tr>
  )
}
export default connect()(ClosedTradeTableRow)