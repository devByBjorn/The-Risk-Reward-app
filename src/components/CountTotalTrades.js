import React from 'react'
import { connect } from 'react-redux'

const ClosedTradesTotal = ({ trades }) => (
  <h2>Total Closed Trades: {trades.length}</h2>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades.filter((trade) => trade.status === 'closed')
  }
}

export default connect(mapStateToProps)(ClosedTradesTotal)