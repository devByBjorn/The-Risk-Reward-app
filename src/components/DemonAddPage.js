import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { addDemon } from '../actions/demonActions'
import DemonFinderForm from './demon-finder/DemonFinderForm'

const AddDemonPage = (props) => (
  <div>
    <h1>Add Demon Page</h1>
    <NavLink to="/demons">Back to demon list</NavLink>
    <div>
      <ol>
        <li>Give your demon suiting a title</li>
        <li>Think about your demon in terms of what,when,why,  how.
          <ul>
            <li>What makes the demon take over?</li>
            <li>When does this demon take over?</li>
            <li>Why...</li>
            <li>How can you manage this demon?</li>
          </ul>
        </li>
        <li>Write down your thoughts and process of managing this demon</li>
      </ol>
      <div>
        <h2>Demon</h2>
        <DemonFinderForm
          handleSubmit={(demon) => {
            props.dispatch(addDemon(demon))
            props.history.push('/demons')
          }}
        />
      </div>
    </div>
  </div>
)

export default connect()(AddDemonPage)