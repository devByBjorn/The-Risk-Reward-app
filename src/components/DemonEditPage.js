import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import DemonFinderForm from '../components/demon-finder/DemonFinderForm'
import { editDemon, deleteDemon } from '../actions/demonActions'
import { TrashIcon } from './table/TradeTableIcons'

class EditDemonPage extends React.Component {
  handleAddEdit = (demon) => {
    const { id } = this.props.demon
    this.props.editDemon(id, demon)
    this.props.history.push('/demons')
  }
  handleDelete = () => {
    const { id } = this.props.demon
    this.props.deleteDemon({ id })
    this.props.history.push('/demons')
  }
  render() {
    return (
      <div>
        <h1>Edit/Read Demon Page</h1>
        <NavLink to="/demons">Back to demon list</NavLink>
        <DemonFinderForm
          demon={this.props.demon}
          handleSubmit={this.handleAddEdit}
        />
        <button
          onClick={this.handleDelete}
        ><TrashIcon />
        </button>
      </div>
    )
  }
}


const mapStateToProps = (state, props) => ({
  demon: state.demons.find((demon) => demon.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch) => ({
  editDemon: (id, demon) => dispatch(editDemon(id, demon)),
  deleteDemon: (id) => dispatch(deleteDemon(id))
})


export default connect(mapStateToProps, mapDispatchToProps)(EditDemonPage)