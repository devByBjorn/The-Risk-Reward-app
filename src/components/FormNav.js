import React, { Fragment } from 'react'
import { Nav, Ul, Li, Button } from '../components_style/formNavStyled'

const FormNav = ({ values, navigateByStepValue }) => {
  const { direction, market, entry, stop, target, setup, status,
    outcome, execution, management, whyExecution, improveExecution,
    whyManagement, improveManagement, step } = values

  const closedSubmitValues = [market, direction, setup, entry, stop,
    target, status, execution, management, whyExecution, improveExecution,
    whyManagement, improveManagement]

  const activePendingSubmitValues = [market, direction, setup, entry, stop,
    target, status]

  const ableSubmitStep = (submitValues) =>
    submitValues.filter((value) =>
      value === '' || value === 0).length !== 0

  return (
    <Nav>
      <Ul>
        <Li>
          <Button
            type="button"
            onClick={() => navigateByStepValue(1)}
            style={step === 1 ? { fontWeight: 'bold' } : {}}
          ><span>Step 1:</span>
            Trade status
            {status === 'closed' && ', dates & outcome'}
            {status === 'active' && ' & open date'}
          </Button>
        </Li>

        {status &&
          <Fragment>
            <Li>
              <Button
                type="button"
                disabled={step < 2 && !market && true}
                onClick={() => navigateByStepValue(2)}
                style={step === 2 ? { fontWeight: 'bold' } : {}}
              ><span>Step 2:</span>
                Market, direction & setup
                </Button>
            </Li>
            <Li>
              <Button
                disabled={step < 3 && !entry && true}
                onClick={() => navigateByStepValue(3)}
                style={step === 3 ? { fontWeight: 'bold' } : {}}
              ><span>Step 3:</span>
              Entry, stop & target
            </Button>
            </Li>
          </Fragment>
        }

        {status === 'closed' &&
          <Fragment>
            <Li>
              <Button
                disabled={step < 4 && !execution && true}
                onClick={() => navigateByStepValue(4)}
                style={step === 4 ? { fontWeight: 'bold' } : {}}
              ><span>Step 4:</span>
                Execution
              </Button>
            </Li>
            <Li>
              <Button
                disabled={step < 5 && !management && true}
                onClick={() => navigateByStepValue(5)}
                style={step === 5 ? { fontWeight: 'bold' } : {}}
              ><span>Step 5:</span>
                Management
              </Button>
            </Li>
            <Li>
              <Button
                disabled={ableSubmitStep(closedSubmitValues) && true}
                onClick={() => navigateByStepValue(6)}
                style={step === 6 ? { fontWeight: 'bold' } : {}}
              ><span>Step 6:</span>
                Inspect & Submit
              </Button>
            </Li>
          </Fragment>
        }

        {status === 'active' &&
          <Fragment>
            <Li>
              <Button
                disabled={ableSubmitStep(activePendingSubmitValues) && true}
                onClick={() => navigateByStepValue(4)}
                style={step === 4 ? { fontWeight: 'bold' } : {}}
              ><span>Step 4:</span>
                Inspect & Submit
              </Button>
            </Li>
            <Li>
              <Button
                disabled={true}
                text=""
                spanText=""
              />
            </Li>
            <Li>
              <Button
                disabled={true}
                text=""
                spanText=""
              />
            </Li>
          </Fragment>
        }

        {status === 'pending' &&
          <Fragment>
            <Li>
              <Button
                disabled={ableSubmitStep(activePendingSubmitValues) && true}
                onClick={() => navigateByStepValue(4)}
                style={step === 4 ? { fontWeight: 'bold' } : {}}
              ><span>Step 4:</span>
                Inspect & Submit
              </Button>
            </Li>
            <Li>
              <Button
                disabled={true}
                text=""
                spanText=""
              />
            </Li>
            <Li>
              <Button
                disabled={true}
                text=""
                spanText=""
              />
            </Li>
          </Fragment>
        }
      </Ul>
    </Nav>
  )
}

export default FormNav
