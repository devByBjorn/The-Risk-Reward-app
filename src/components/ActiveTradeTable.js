import React from 'react'
import { connect } from 'react-redux'
import ActiveTableRow from './ActiveTableRow'
import { ThMobileAndDesktop, ThOnlyDesktop } from './TradeTableHead'
const ActiveTradeTable = ({ trades }) => (
  <div>
    <table className="table-grid active">
      <caption>Active Trades</caption>
      <thead>
        <tr>
          <ThMobileAndDesktop heading='Market' />
          <ThMobileAndDesktop heading='Direction' />
          <ThMobileAndDesktop heading='Setup' />
          <ThMobileAndDesktop heading='Entry' />
          <ThMobileAndDesktop heading='Stop' />
          <ThMobileAndDesktop heading='Target' />
          <ThOnlyDesktop heading='Opened' />
          <ThMobileAndDesktop heading='-R' />
          <ThMobileAndDesktop heading='+R' />
        </tr>
      </thead>
      <tbody>
        {trades.map((trade) =>
          (<ActiveTableRow
            {...trade}
            key={trade.id}
          />)
        )}
      </tbody>
    </table>
  </div>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades,
  }
}

export default connect(mapStateToProps)(ActiveTradeTable)