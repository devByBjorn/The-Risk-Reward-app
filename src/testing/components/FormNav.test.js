import React from 'react'
import { shallow } from 'enzyme'
import { FormNav } from '../../components/FormNav'

const defaultValues = {
  step: 0,
  direction: '',
  market: '',
  entry: 0,
  stop: 0,
  target: 0,
  exit: 0,
  status: '',
  setup: '',
  opened: 0,
  closed: 0,
  period: 0,
  rewardToRisk: 0,
  rMultiple: 0,
  risk: 0,
  reward: 0,
  conclusion: {
    execution: '',
    improveExecution: '',
    management: '',
    improveManagement: '',
    whyExecution: '',
    whyManagement: '',
  }
}

test('should render FormNav without content', () => {
  const wrapper = shallow(
    <FormNav
      values={defaultValues}
    />
  )
  expect(wrapper).toMatchSnapshot()
})


// STEP 1 
test('should call navigateStepValue with 1 on click', () => {
  const navigateByStepValue = jest.fn()
  const values = { step: 1, status: true }
  const wrapper = shallow(
    <FormNav
      navigateByStepValue={navigateByStepValue}
      values={values}
    />
  )

  wrapper.find('FormNavStyles__Button').at(0).simulate('click')
  expect(navigateByStepValue).toHaveBeenCalledWith(1)
})

// STEP 2  
test('should call navigateStepValue with 2 on click', () => {
  const navigateByStepValue = jest.fn()
  const values = { step: 2, status: true }
  const wrapper = shallow(
    <FormNav
      navigateByStepValue={navigateByStepValue}
      values={values}
    />
  )

  wrapper.find('FormNavStyles__Button').at(1).simulate('click')
  expect(navigateByStepValue).toHaveBeenCalledWith(2)
})


// STEP 3 
test('should call navigateStepValue with 3 on click', () => {
  const navigateByStepValue = jest.fn()
  const values = { step: 3, status: true }
  const wrapper = shallow(
    <FormNav
      navigateByStepValue={navigateByStepValue}
      values={values}
    />
  )

  wrapper.find('FormNavStyles__Button').at(2).simulate('click')
  expect(navigateByStepValue).toHaveBeenCalledWith(3)
})

//  CLOSED STATUS
test('should render FormNav for closed status', () => {
  const values = { status: 'closed' }
  const wrapper = shallow(
    <FormNav
      values={values}
    />
  )
  expect(wrapper).toMatchSnapshot()
})

// STEP 4  CLOSED STATUS
test('should call navigateStepValue with 4 on click', () => {
  const navigateByStepValue = jest.fn()
  const values = { step: 4, status: 'closed' }
  const wrapper = shallow(
    <FormNav
      navigateByStepValue={navigateByStepValue}
      values={values}
    />
  )

  wrapper.find('FormNavStyles__Button').at(3).simulate('click')
  expect(navigateByStepValue).toHaveBeenCalledWith(4)
})


// STEP 5  CLOSED STATUS
test('should call navigateStepValue with 5 on click', () => {
  const navigateByStepValue = jest.fn()
  const values = { step: 5, status: 'closed' }
  const wrapper = shallow(
    <FormNav
      navigateByStepValue={navigateByStepValue}
      values={values}
    />
  )

  wrapper.find('FormNavStyles__Button').at(4).simulate('click')
  expect(navigateByStepValue).toHaveBeenCalledWith(5)
})

// STEP 6  CLOSED STATUS
test('should call navigateStepValue with 6 on click', () => {
  const navigateByStepValue = jest.fn()
  const values = { step: 6, status: 'closed' }
  const wrapper = shallow(
    <FormNav
      navigateByStepValue={navigateByStepValue}
      values={values}
    />
  )

  wrapper.find('FormNavStyles__Button').at(5).simulate('click')
  expect(navigateByStepValue).toHaveBeenCalledWith(6)
})

// ACTIVE STATUS
test('should render FormNav for Active status', () => {
  const values = { status: 'active' }
  const wrapper = shallow(
    <FormNav
      values={values}
    />
  )
  expect(wrapper).toMatchSnapshot()
})

// STEP 4 
test('should call navigateStepValue with 4 on click', () => {
  const navigateByStepValue = jest.fn()
  const values = { step: 4, status: 'active' }
  const wrapper = shallow(
    <FormNav
      navigateByStepValue={navigateByStepValue}
      values={values}
    />
  )

  wrapper.find('FormNavStyles__Button').at(3).simulate('click')
  expect(navigateByStepValue).toHaveBeenCalledWith(4)
})

// PENDING STATUS
test('should render FormNav for Pending status', () => {
  const values = { status: 'pending' }
  const wrapper = shallow(
    <FormNav
      values={values}
    />
  )
  expect(wrapper).toMatchSnapshot()
})

// STEP 4 
test('should call navigateStepValue with 4 on click', () => {
  const navigateByStepValue = jest.fn()
  const values = { step: 4, status: 'pending' }
  const wrapper = shallow(
    <FormNav
      navigateByStepValue={navigateByStepValue}
      values={values}
    />
  )

  wrapper.find('FormNavStyles__Button').at(3).simulate('click')
  expect(navigateByStepValue).toHaveBeenCalledWith(4)
})