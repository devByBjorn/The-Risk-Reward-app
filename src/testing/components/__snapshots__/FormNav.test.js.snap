// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormNav for Active status 1`] = `
<FormNavStyles__Nav>
  <FormNavStyles__Ul>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Step 1:
        </span>
        Status
         & open date
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Step 2:
        </span>
        Market, direction & setup
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          Step 3:
        </span>
        Entry, stop & target
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          Step 4:
        </span>
        Inspect & Submit
      </FormNavStyles__Button>
    </FormNavStyles__Li>
  </FormNavStyles__Ul>
</FormNavStyles__Nav>
`;

exports[`should render FormNav for Pending status 1`] = `
<FormNavStyles__Nav>
  <FormNavStyles__Ul>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Step 1:
        </span>
        Status
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Step 2:
        </span>
        Market, direction & setup
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          Step 3:
        </span>
        Entry, stop & target
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          Step 4:
        </span>
        Inspect & Submit
      </FormNavStyles__Button>
    </FormNavStyles__Li>
  </FormNavStyles__Ul>
</FormNavStyles__Nav>
`;

exports[`should render FormNav for closed status 1`] = `
<FormNavStyles__Nav>
  <FormNavStyles__Ul>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Step 1:
        </span>
        Status
        , dates & outcome
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span>
          Step 2:
        </span>
        Market, direction & setup
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          Step 3:
        </span>
        Entry, stop & target
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          Step 4:
        </span>
        Execution
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          Step 5:
        </span>
        Management
      </FormNavStyles__Button>
    </FormNavStyles__Li>
    <FormNavStyles__Li>
      <FormNavStyles__Button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          Step 6:
        </span>
        Inspect & Submit
      </FormNavStyles__Button>
    </FormNavStyles__Li>
  </FormNavStyles__Ul>
</FormNavStyles__Nav>
`;

exports[`should render FormNav without content 1`] = `
<FormNavStyles__Nav>
  <FormNavStyles__Ul />
</FormNavStyles__Nav>
`;
